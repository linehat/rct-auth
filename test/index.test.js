import React from "react";
import _ from "lodash";
import authHOC from "../src/";
import renderer from "react-test-renderer";

it("render auth", () => {
  const Div = props => <div>{props.children}</div>;
  const AuthDiv = authHOC(Div, id => _.includes(["1", "2", "3"], id));
  const tree = renderer
    .create(
      <div>
        <AuthDiv authId="1"><h2>auth ok</h2></AuthDiv>
        <AuthDiv authId="0"><h2>no auth</h2></AuthDiv>
        <AuthDiv authId="1" noCached><h2>no auth</h2></AuthDiv>
        <AuthDiv authId="0" noCached><h2>no auth</h2></AuthDiv>
      </div>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
