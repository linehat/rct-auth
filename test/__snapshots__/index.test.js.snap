// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render auth 1`] = `
<div>
  <div>
    <h2>
      auth ok
    </h2>
  </div>
  <div>
    <h2>
      no auth
    </h2>
  </div>
</div>
`;
